src/main/java/pl/onsight/wypozyczalnia/controller/UserRegisterController.java

```diff
@@ -30,13 +39,21 @@ public ModelAndView registrationPage(ModelAndView modelAndView) {
     }
 
     @PostMapping("/register")
-    public String addUser(@ModelAttribute @Valid UserEntity user,
-                          BindingResult bindResult) {
+    public ModelAndView addUser(@ModelAttribute @Valid UserEntity user,
+                                BindingResult bindResult, ModelAndView modelAndView) {
         if (bindResult.hasErrors()) {
-            return "register";
+            modelAndView.setViewName("register");
+        } else if (registerValidator.isEmailTaken(user)) {
+            modelAndView.addObject("emailIsTaken", "Użytkownik o podanym emailu juz istnieje!");
```

Comment:
nie wiem czy jest sens rozdzielac to info na 2 osobne

Reply:
Ładniej to wyglada. ale jak cos to zmienie ;D to tak pierdola.